pandas/core/indexes/base.py

```diff
@@ -1012,6 +1012,16 @@ def view(self, cls=None):
 
             result = self._data.view(cls)
         else:
+            if cls is not None:
```

Comment:
There is still a `if cls is not None and not hasattr(cls, "_typ"):` branch above, should passing `cls` for those cases also be deprecated? 
(otherwise we still can't remove the keyword after the current deprecation of this PR is enforced?)

Reply:
OK, I see. But do you know of any usage internally with a dtype? 
I know we use it regularly without any argument to get an "identical" object (that passes `is_`), but I don't directly find much usage with a dtype.

From a quick look (outside of direct tests for this), I only found `CategoricalDtype._hash_categories` that uses it, where it could easily be avoided.